### ISSUE TITLE
Zip file extracted permissions are incorrect

### ISSUE THREAD

**Issue by sidcarter** [saltstack/salt#23822]
I'm using this state to install packer:

```
{% set version = '0.7.5' %}

packer:
  archive:
    - extracted
    - name: /usr/local/bin/
    - source: https://dl.bintray.com/mitchellh/packer/packer_{{version}}_linux_amd64.zip
    - source_hash: sha256=8fab291c8cc988bd0004195677924ab6846aee5800b6c8696d71d33456701ef6
    - archive_format: zip
    - if_missing: /usr/local/bin/packer
```

This state command itself worked fine without any issues, but then I noticed that the permissions were all wrong. 
```
# salt --version
salt 2015.5.0 (Lithium)
```
```
# ls -l
total 318964
-rw-r--r-- 1 root root 10087048 May 17 05:47 packer
-rw-r--r-- 1 root root 10293560 May 17 05:47 packer-builder-amazon-chroot
-rw-r--r-- 1 root root 11110464 May 17 05:47 packer-builder-amazon-ebs
-rw-r--r-- 1 root root 11146448 May 17 05:47 packer-builder-amazon-instance
-rw-r--r-- 1 root root 10437344 May 17 05:47 packer-builder-digitalocean
-rw-r--r-- 1 root root  9394712 May 17 05:47 packer-builder-docker
```

If I extracted the zip directly, I get this:

```
~/test# ls -l
total 318964
-rwxr-xr-x 1 root root 10087048 Dec 10 03:17 packer
-rwxr-xr-x 1 root root 10293560 Dec 10 03:17 packer-builder-amazon-chroot
-rwxr-xr-x 1 root root 11110464 Dec 10 03:17 packer-builder-amazon-ebs
-rwxr-xr-x 1 root root 11146448 Dec 10 03:17 packer-builder-amazon-instance
-rwxr-xr-x 1 root root 10437344 Dec 10 03:17 packer-builder-digitalocean
-rwxr-xr-x 1 root root  9394712 Dec 10 03:17 packer-builder-docker
```
I don't know of any options I can pass to fix this. Is this a bug?

**Comment by jfindlay** [comment]
@sidcarter, thanks for the report.

**Comment by sebest** [comment]
This is a bug in python zipfile lib with extractall()
Previous version of salt was using bin/unzip 
Salt should revert to using bin/unzip as default

**Comment by jfindlay** [comment]
@sebest, do you have a reference to the upstream bug?  Does one need to be filed? :-)

**Comment by jfindlay** [comment]
This might be it: https://bugs.python.org/issue15795

**Comment by sebest** [comment]
Any update on this?

**Comment by sebest** [comment]
Hi, i hoped that this got fixed in 2015.5.3 , but it seems that it did not make it?

**Comment by jfindlay** [comment]
The fix (https://github.com/saltstack/salt/pull/25185) missed 2015.5.3 by 50 commits, (`git describe 1726057c8a93101b58c8e2cdb5d0d5b021dc29a6 -> v2015.5.3-50-g1726057`), but will be in 2015.5.4 or 2015.8.0, whichever comes out first.

**Comment by rallytime** [comment]
This should be fixed in the latest stable release, 2015.5.5. As such, I am going to close this issue. If this pops up again, let us know and we will be happy to re-open. Thanks!
